realm/src/main/java/io/realm/DynamicRealm.java

```diff
@@ -0,0 +1,283 @@
+     */
+    public DynamicRealmObject createObject(String className) {
+        checkIfValid();
+        Table table = getTable(className);
+        long rowIndex = table.addEmptyRow();
+        return get(DynamicRealmObject.class, className, rowIndex);
+    }
+
+    /**
+     * Returns a RealmQuery, which can be used to query for the provided class.
+     *
+     * @param className The class of the object which is to be queried for.
+     * @return A RealmQuery, which can be used to query for specific objects of provided type.
+     * @see io.realm.RealmQuery
+     * @throws IllegalArgumentException if the class doesn't exist.
+     */
+    public RealmQuery<DynamicRealmObject> where(String className) {
+        checkIfValid();
+        if (!sharedGroupManager.hasTable(Table.TABLE_PREFIX + className)) {
+            throw new IllegalArgumentException("Class does not exist in the Realm so it cannot be queried: " + className);
+        }
+        return RealmQuery.createDynamicQuery(this, className);
+    }
+
+    /**
+     * Removes all objects of the specified class.
+     *
+     * @param className The class for which all objects should be removed.
+     */
+    public void clear(String className) {
+        checkIfValid();
+        getTable(className).clear();
+    }
+
+    /**
+     * Executes a given transaction on the DynamicRealm. {@link #beginTransaction()} and
+     * {@link #commitTransaction()} will be called automatically. If any exception is thrown
+     * during the transaction {@link #cancelTransaction()} will be called instead of {@link #commitTransaction()}.
+     *
+     * @param transaction {@link io.realm.DynamicRealm.Transaction} to execute.
+     * @throws RealmException if any error happened during the transaction.
+     */
+    public void executeTransaction(Transaction transaction) {
+        if (transaction == null)
+            return;
+        beginTransaction();
+        try {
+            transaction.execute(this);
+            commitTransaction();
+        } catch (RuntimeException e) {
+            cancelTransaction();
+            throw new RealmException("Error during transaction.", e);
+        } catch (Error e) {
+            cancelTransaction();
+            throw e;
+        }
+    }
+
+    /**
+     * Get all objects of a specific class name.
+     *
+     * @param className the Class to get objects of
+     * @return A RealmResult list containing the objects. If no results where found, an empty list
+     * will be returned.
+     * @see io.realm.RealmResults
+     */
+    public RealmResults<DynamicRealmObject> allObjects(String className) {
+        return where(className).findAll();
+    }
+
+    /**
+     * Get all objects of a specific class name sorted by a field.  If no objects exist, the returned
+     * RealmResults will not be null. Use {@link RealmResults#size()} to check the number of objects instead.
+     *
+     * @param className The class to get all objects from.
```

Comment:
Yup, this will have to be adjusted all over the place. The Javadoc was written before we decided on the change, and I forgot to update it :(